dqnroute/simpy_router.py

```diff
@@ -0,0 +1,57 @@
+import simpy
+import random
+
+class SimPyAbstractRouter(object):
+    def __init__(self, env, id):
+        self.env = env
+        self.id = id
+        
+        self.send_process = env.process(self.sendPkgProcess())
+        self.get_process = env.process(self.getPkgProcess())
+        self.send_event = env.event()
+        self.get_event = env.event()
+
+    def setNeighbours(self, neighbours):
+        self.nghbrs = neighbours
+        
+    def sendPkgProcess(self):
+        while True:
+            pkg, dest = yield self.send_event
+            destRouter = self.nghbrs[dest]
```

Comment:
1) `nghbrs` is an array of adjacent routers.
2) You use the ID of the destination node as an index in this array.
3) If your code does not get an `IndexError` somehow, then it most likely routes elsewhere but the destination router, because how on Earth indices in this array and node IDs are connected?

Also, this code ignores the case when the destination router is the current one.